examples/trusted-sync/src/main.rs

```diff
@@ -71,9 +72,14 @@ async fn sync(cli: cli::Cli) -> Result<()> {
     let mut l2_provider = AlloyL2ChainProvider::new_http(l2_rpc_url.clone(), cfg.clone());
     let attributes =
         StatefulAttributesBuilder::new(cfg.clone(), l2_provider.clone(), l1_provider.clone());
-    let beacon_client = OnlineBeaconClient::new_http(beacon_url);
-    let blob_provider =
-        OnlineBlobProvider::<_, SimpleSlotDerivation>::new(beacon_client, None, None);
+    let blob_provider = if let Some(blob_archiver_url) = blob_archiver_url {
+        OnlineBlobProviderBuilder::new()
+            .with_primary_beacon_client_url(beacon_url)
+            .with_fallback_beacon_client_url(blob_archiver_url)
+            .build()
+    } else {
+        OnlineBlobProviderBuilder::new().with_primary_beacon_client_url(beacon_url).build()
+    };
```

Comment:
Can we just have `with_fallback_beacon_client_url` accept an Option here?

Then we can just do:
```rust
let blob_provider = OnlineBlobProviderBuilder::new().with_primary(beacon_url).with_fallback(cli.blob_archiver_url()).build()

```